spec/useRenderCampaigns.spec.tsx

```diff
@@ -0,0 +1,115 @@
+import { act } from "react-dom/test-utils"
+import { useRenderCampaigns } from "../src/hooks/useRenderCampaigns"
+import { renderHook } from "@testing-library/react"
+import { describe, beforeEach, it, expect, vi } from "vitest"
+import RecommendationComponent from "./renderer"
+import { ReactNode } from "react"
+import { NostoContext } from "../src/context"
+
+describe("useRenderCampaigns", () => {
+
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <div id="frontpage-nosto-1"></div>
+      <div id="frontpage-nosto-2"></div>`
+  })
+
+  it("supports component rendering", async () => {
+    function Wrapper({ children }: { children: ReactNode }) {
```

Comment:
Could be extracted into a utility function. It feels like this would be useful in the future tests as well.